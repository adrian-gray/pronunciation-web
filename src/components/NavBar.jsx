import React from 'react'
import { Link } from 'react-router-dom'
import firebase from 'firebase/app'

import {
  AppBar,
  Button,
  Drawer,
  MenuItem,
  Toolbar,
  Typography,
  withStyles
} from '@material-ui/core'

const styles = (theme) => ({
  root: {
    flexGrow: 1
  },
  flex: {
    flexGrow: 1
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20
  },
  list: {
    width: 250
  },
  white: {
    color: '#FFFFFF'
  }
})

class NavBar extends React.Component {
  constructor (props) {
    super(props)

    this.state = {
      isLoggedIn: false,
      open: false
    }

    this.handleToggle = this.handleToggle.bind(this)
    this.initFirebaseAuth = this.initFirebaseAuth.bind(this)
    this.handleAuthChange = this.handleAuthChange.bind(this)
  }

  componentDidMount () {
    this.initFirebaseAuth()
  }

  handleToggle () {
    this.setState({ open: !this.state.open })
  }

  initFirebaseAuth () {
    firebase.auth().onAuthStateChanged(this.handleAuthChange)
  }

  handleAuthChange (e) {
    let isLoggedIn = false
    if (e) {
      isLoggedIn = true
    }

    this.setState({ isLoggedIn })
  }

  handleGoogleLogin () {
    const provider = new firebase.auth.GoogleAuthProvider()
    firebase.auth().signInWithPopup(provider)
  }

  handleSignOut () {
    firebase.auth().signOut()
  }

  render () {
    const classes = this.props.classes

    let changeLoginStatus
    console.log('render', this.state.isLoggedIn)
    if (this.state.isLoggedIn) {
      changeLoginStatus = (
        <Button onClick={this.handleSignOut} color='inherit'>{'Sign Out'}</Button>
      )
    } else {
      changeLoginStatus = (
        <Button component={Link} to='/login' color='inherit'>{'Log In'}</Button>
      )
    }

    return (
      <div className={classes.root}>
        <AppBar position='static'>
          <Toolbar>
            <Drawer
              open={this.state.open}
              onClick={this.handleToggle}
              onKeyDown={this.handleToggle}
            >
              <MenuItem>
                <Button href='/' className={classes.button}>
                  {'Home'}
                </Button>
              </MenuItem>
              <MenuItem>
                <Button href='/sounds' className={classes.button}>
                  {'Sounds'}
                </Button>
              </MenuItem>
            </Drawer>
            <Typography
              variant='subheading'
              className={classes.flex}
            >
              <Button href='/'>
                <span className={classes.white}>
                  {'Pronounce Web'}
                </span>
              </Button>
            </Typography>
            {changeLoginStatus}
          </Toolbar>
        </AppBar>
      </div>
    )
  }
}

export default withStyles(styles)(NavBar)
