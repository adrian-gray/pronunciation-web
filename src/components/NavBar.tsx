import React from "react";

import Navbar from "react-bootstrap/Navbar";
import Button from "react-bootstrap/Button";
import Form from "react-bootstrap/Form";
import Nav from "react-bootstrap/Nav";

export default props => {
  let homeLink = "/";
  let changeLoginStatus;

  if (props.user) {
    changeLoginStatus = (
      <Form inline>
        <Button variant="outline-info" onClick={props.signout} color="inherit">
          {"Sign Out"}
        </Button>
      </Form>
    );
    homeLink = "/home";
  } else {
    changeLoginStatus = (
      <Form inline>
        <Button href="/login" color="inherit">
          {"Sign In or Up"}
        </Button>
      </Form>
    );
  }

  return (
    <Navbar bg="dark" variant="dark" className="justify-content-between">
      <Navbar.Brand href={homeLink}>Pronounce Web</Navbar.Brand>
      <Navbar.Toggle aria-controls="basic-navbar-nav" />
      <Navbar.Collapse className="justify-content-end">
        <Nav>{changeLoginStatus}</Nav>
      </Navbar.Collapse>
    </Navbar>
  );
};
