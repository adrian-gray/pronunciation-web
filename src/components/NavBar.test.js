import React from "react";
import { shallow } from "enzyme";
import { render, cleanup } from "@testing-library/react";

import NavBar from "./NavBar";

afterEach(cleanup);

test("NavBar", async () => {
  const props = {};

  const container = shallow(<NavBar {...props} />);

  expect(container.firstChild).toMatchSnapshot();
});
