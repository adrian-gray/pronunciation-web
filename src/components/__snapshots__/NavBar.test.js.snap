// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar 1`] = `
<div>
  <nav
    class="justify-content-between navbar navbar-expand navbar-dark bg-dark"
  >
    <a
      class="navbar-brand"
      href="/home"
    >
      Pronounce Web
    </a>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      class="navbar-toggler"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="justify-content-end navbar-collapse collapse"
    >
      <div
        class="navbar-nav"
      >
        <form
          class="form-inline"
        >
          <button
            class="btn btn-outline-info"
            color="inherit"
            type="button"
          >
            Sign Out
          </button>
        </form>
      </div>
    </div>
  </nav>
</div>
`;
